src/Solr/QueryBuilder/Update.php

```diff
@@ -14,14 +14,16 @@
  */
 class Update extends AbstractQueryBuilder
 {
+    private const STREAM_MAX_MEMORY = 50 * 1048576; // 50 Mb, default is 2 Mb
+
     private $stream = null;
 
     /**
      * Initialize the update request.
      */
     public function __construct()
     {
-        $this->stream = fopen('php://temp', 'rw');
+        $this->stream = fopen(sprintf('php://temp/maxmemory:%s', self::STREAM_MAX_MEMORY), 'rw');
         fwrite($this->stream, '{');
```

Comment:
Dit ziet er uit alsof dit 12 jaar geleden noodzakelijk was, is er inmiddels geen modernere manier om een update naar de SOLR-server te sturen dan het aan elkaar plakken van stukjes tekst?

Reply:
Hmm... dit verzamelt de data in memory en na 50Mb naar disk. Dus dat is wel handig als het honderden Mb's worden. Maar gewoon door PHP de data in een array verzamelen en dan `json_encode` erover en klaar, dat kan ook gewoon lijkt mij ja 😄